src/execution_plan/ops/op_update.c

```diff
@@ -128,38 +198,33 @@ Record OpUpdateConsume(OpBase *opBase) {
     OpUpdate *op = (OpUpdate*)opBase;
     OpBase *child = op->op.children[0];
     Record r = child->consume(child);
+    // No data, return.
     if(!r) return NULL;
 
     /* Evaluate each update expression and store result 
      * for later execution. */
     EntityUpdateEvalCtx *update_expression = op->update_expressions;
     for(int i = 0; i < op->update_expressions_count; i++, update_expression++) {
         SIValue new_value = AR_EXP_Evaluate(update_expression->exp, r);
-        /* Find ref to property. */
         GraphEntity *entity = Record_GetGraphEntity(r, update_expression->entityRecIdx);
-        int j = 0;
-        for(; j < ENTITY_PROP_COUNT(entity); j++) {
-            if(strcmp(ENTITY_PROPS(entity)[j].name, update_expression->property) == 0) {
-                _OpUpdate_QueueUpdate(op, entity->entity, update_expression->ge,
-                                      j, new_value);
-                break;
-            }
-        }
-
-        if(j == ENTITY_PROP_COUNT(entity)) {
-            /* Property does not exists for entity, create it.
-             * For the time being set the new property value to PROPERTY_NOTFOUND.
-             * Once we commit the update, we'll set the actual value. */
-            GraphEntity_Add_Properties(entity, 1, &update_expression->property, PROPERTY_NOTFOUND);
-            _OpUpdate_QueueUpdate(op, entity->entity, update_expression->ge,
-                                  ENTITY_PROP_COUNT(entity)-1, new_value);
-        }
+        _QueueUpdate(op,
+                     entity->entity,
+                     update_expression->ge,
+                     update_expression->attribute,
+                     update_expression->attribute_idx,
+                     new_value);
     }
 
     return r;
 }
 
 OpResult OpUpdateReset(OpBase *ctx) {
+    OpUpdate *op = (OpUpdate*)ctx;
+    // Reset all pending updates.
+    rm_free(op->pending_updates);
+    op->pending_updates_count = 0;
+    op->pending_updates_cap = 16; /* 16 seems reasonable number to start with. */
+    op->pending_updates = rm_malloc(sizeof(EntityUpdateCtx) * op->pending_updates_cap);
```

Comment:
Why not realloc instead of freeing then mallocing?

Reply:
At first my thought was to avoid realloc memcpy, but now I hope that when reallocing to the exact same size or to a smaller size there's no need for memcpy.